transport/owr_transport_agent.c

```diff
@@ -470,6 +475,10 @@ static void add_helper_server_info(GResolver *resolver, GAsyncResult *result, GH
     g_return_if_fail(OWR_IS_TRANSPORT_AGENT(transport_agent));
     g_hash_table_remove(info, "transport_agent");
 
+    priv = transport_agent->priv;
+
+    priv->deferred_helper_server_adds--;
+
     address_list = g_resolver_lookup_by_name_finish(resolver, result, &error);
 
     if (!address_list) {
```

Comment:
I think in the resolval failure case you also should check if deferred_helper_server_adds is 0 and if so call gather_candidates(). Otherwise if there are DNS failures we might never call it